lib/mongo/retryable/base_worker.rb

```diff
@@ -58,7 +58,8 @@ def retryable_exceptions
           Error::ConnectionPerished,
           Error::ServerNotUsable,
           Error::SocketError,
-          Error::SocketTimeoutError
+          Error::SocketTimeoutError,
+          Error::PoolError,
```

Comment:
```suggestion
          Error::PoolClearedError,
```
According to the [specification](https://github.com/mongodb/specifications/blob/663db4db13c8acf415bb97259bf7f1909b32a670/source/retryable-reads/retryable-reads.md#retryable-error) only the `PoolClearedError` should be retried, not all pool errors. So, having `PoolError` here seems to be too broad.

Reply:
@comandeo-mongo `Error::ServerNotUsable` and `Error::ConnectionPerished` are listed here as retryable but not in the specification, could you clarify if they are retryable? Asking because we started seeing these errors after upgrading to the latest.

Reply:
hmm, the [code](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/retryable/read_worker.rb#L216) currently seems to retry on any pool errors

Reply:
@comandeo-mongo I've addressed your comment, please look at it when you have the time.

Reply:
Hi @joelim41 , I am finally looking into it, sorry for the delay. Meanwhile, I would like to clarify - did you start seeing `Error::ServerNotUsable` and `Error::ConnectionPerished` only after the upgrade? It might be an indication of some other issues.

Reply:
@joelim41 one more "side" question - is there something tpreventing you from using modern retries? Legacy retry mechanism is deprecates since `2.9.0`

Reply:
Oh, I missed that you are upgrading from 2.16. So yes, you weren't able to see those errors. I think it is OK if you see them when a mongos has issues.

As to this PR - I think we should separate the list of the retryable errors for modern and legacy retries. `PoolError` has a special treatment in modern retries, so I would like to avoid just adding it to `retryable_exceptions`. I'll make a commit to your PR with the proposed solution.

Reply:
Got it, thank you!